lua/obsidian/command.lua

```diff
@@ -294,6 +294,62 @@ command.complete_args = function(client, arg_lead, cmd_line, cursor_pos)
   return completions
 end
 
+--- Follow link under cursor.
+---
+---@param client obsidian.Client
+command.follow = function(client, _)
+  local scan = require "plenary.scandir"
+
+  local open, close = util.cursor_on_markdown_link()
+  local current_line = vim.api.nvim_get_current_line()
+
+  if open == nil or close == nil then
+    return nil
+  end
+
+  local note_name = current_line:sub(open + 2, close - 1)
+  local path = client.dir
+  
+  if note_name:match("|[^%]]*") then
+    note_name = note_name:sub(1, note_name:find('|'))
+  end
+
+  if not note_name:match("%.md") then
+    note_name = note_name .. ".md"
+  end
+  
+
+  local notes = {}
+
+  if not note_name:match("/") then
+    scan.scan_dir(vim.fs.normalize(tostring(client.dir)), {
+      hidden = false,
+      add_dirs = false,
+      only_dirs = true,
+      respect_gitignore = true,
+      on_insert = function(entry)
+        note_dir = Path:new(entry .. "/" .. note_name)
+        if note_dir:is_dir() then
+          local _, note = pcall(Note.from_file, entry .. "/" .. note_name, client.dir)
+          table.insert(notes, note.path)
```

Comment:
`note` could be `nil` if this fails, so I think you need:

```suggestion
          local ok, note = pcall(Note.from_file, entry .. "/" .. note_name, client.dir)
          if ok then
            table.insert(notes, note.path)
          end
```